src/main/java/com/hazelcast/web/WebFilter.java

```diff
@@ -512,8 +517,9 @@ private String findHazelcastSessionIdFromRequest() {
                     }
                 }
             }
-            // if hazelcast session id is not found on the cookie, look into request parameters
-            if (hzSessionId == null) {
+            // if hazelcast session id is not found on the cookie and using request parameter is enabled, look into
+            // request parameters
+            if (hzSessionId == null && useRequestParameter) {
```

Comment:
so from now on, we will only look into request parameters for session id in case user enables `useRequestParameter` in web.xml. That can affect existing people who upgrade to the latest version. What if we make `useRequestParameter=true` by default and ask developers to disable it explicitly if they are having issues?